picard/ui/options/plugins.py

```diff
@@ -230,6 +230,25 @@ def plugin_updated(self, plugin_name):
                 self.change_details()
                 break
 
+    def plugin_remove(self):
+            plugin = self.items[self.ui.plugins.selectedItems()[0]]
+            buttonReply = QtWidgets.QMessageBox.question(self,'Uninstall?', "Do you want to uninstall?",
+                    QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No, QtWidgets.QMessageBox.No)
+            if buttonReply == QtWidgets.QMessageBox.Yes:
+                if os.path.isfile(USER_PLUGIN_DIR+plugin.module_name+'.zip'):
```

Comment:
I cannot fathom why you need to add extensions to the plugin names. It will only cause erroneous results with remove_plugin. Remove_plugin just expects the plugin name by default. Nothing else. It handles stripping of extensions and deletion of files.

See https://github.com/metabrainz/picard/blob/master/picard/plugin.py#L351

Reply:
I think we can do something like this.
<img width="622" alt="screen shot 2018-01-21 at 7 17 28 pm" src="https://user-images.githubusercontent.com/20679490/35194827-b6c78d7c-fedf-11e7-8c4b-0de8fbc8d50a.png">
Instead of course, following=> uninstall and follow=>install. The important thing here is to keep it looking like a button so that people know it can be clicked.

Reply:
@vishichoudhary 
1. I don't know the case here, but if update is available for only a few plugins at a time,
You can add an "update" link in front of all plugins whose update is available.

2. If an update is available for all the plugins all the time, then having install/uninstall button and an update button makes more sense.

Reply:
@chhavijustme  what about having only one button for install / uninstall and update automatically?